### PR TITLE
Use extension to detect itc errors

### PR DESCRIPTION
**Pull request by cquiroz**

### CODE REVIEW

**Review by toddburnside — APPROVED** [review]
LGTM

File: modules/service/src/main/scala/lucuma/odb/service/ItcService.scala [hunk 1]
```diff
@@ -388,7 +389,7 @@ object ItcService {
         client.imaging(ii, useCache = false)
           .map(_.result)
           .recover {
-            case ResponseException(errors, _) if errors.exists(_.message.contains("target is too bright")) =>
+            case ResponseException(errors, _) if errors.exists(_.extensions.exists(_.exists(_ === ("errorCode" -> ErrorCode.SourceTooBright.asJson)))) =>
```
Comment (swalker2m) [inline modules/service/src/main/scala/lucuma/odb/service/ItcService.scala:392]: If there were other errors, besides the source-too-bright one, would we still want to use the default integration time?

**Review by swalker2m — APPROVED** [review]
Looks good.  :+1:  If we do this much, we might want some clue syntax or something to simplify.

File: modules/service/src/main/scala/lucuma/odb/service/ItcService.scala (hunk 1 above)
Comment (cquiroz) [reply modules/service/src/main/scala/lucuma/odb/service/ItcService.scala:392]: Good question. I'd say only this one, if e.g. the config is unsupported you ddon't really want to default to a valid value

**Pull request merged**